src/main/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializer.java

```diff
@@ -30,50 +31,60 @@ public class RandomTxPowerInitializer extends TPC {
 	/** The RRM algorithm ID. */
 	public static final String ALGORITHM_ID = "random";
 
-	/** The fixed tx power (dBm). */
-	private final int txPower;
+	/** The PRNG instance. */
+	private Random rng = new Random();
 
-	/** Constructor (uses random tx power). */
-	public RandomTxPowerInitializer(
-		DataModel model, String zone, DeviceDataManager deviceDataManager
-	) {
-		this(model, zone, deviceDataManager, getRandomTxPower());
-	}
+	/** Whether to set a different value per AP or use a single value for all APs */
+	private boolean setDifferentTxPowerPerAp = false;
 
-	/** Constructor. */
+	/** Constructor (uses random tx power per AP and allows passing in a custom Random class to allow seeding). */
 	public RandomTxPowerInitializer(
 		DataModel model,
 		String zone,
 		DeviceDataManager deviceDataManager,
-		int txPower
+		boolean setDifferentTxPowerPerAp,
+		Random rng
+	) {
+		this(model, zone, deviceDataManager, setDifferentTxPowerPerAp);
+		this.rng = rng;
+	}
+
+	/** Constructor (uses random tx power per AP). */
+	public RandomTxPowerInitializer(
+		DataModel model, String zone, DeviceDataManager deviceDataManager, boolean setDifferentTxPowerPerAp
+	) {
+		this(model, zone, deviceDataManager);
+		this.setDifferentTxPowerPerAp = setDifferentTxPowerPerAp;
+	}
+
+	/** Constructor (uses random tx power). */
+	public RandomTxPowerInitializer(
+		DataModel model, String zone, DeviceDataManager deviceDataManager
 	) {
 		super(model, zone, deviceDataManager);
-		this.txPower = txPower;
 	}
 
 	/** Get a random tx power in [MIN_TX_POWER, MAX_TX_POWER], both inclusive */
-	public static int getRandomTxPower() {
-		Random rand = new Random();
-		return rand.nextInt(TPC.MAX_TX_POWER + 1 - TPC.MIN_TX_POWER) + TPC.MIN_TX_POWER;
+	public int getRandomTxPower() {
+		return rng.nextInt(TPC.MAX_TX_POWER + 1 - TPC.MIN_TX_POWER) + TPC.MIN_TX_POWER;
 	}
 
 	@Override
 	public Map<String, Map<String, Integer>> computeTxPowerMap() {
+		int defaultTxPower = this.getRandomTxPower();
+		logger.info("Default power: {}", defaultTxPower);
 		Map<String, Map<String, Integer>> txPowerMap = new TreeMap<>();
 		for (String serialNumber : model.latestState.keySet()) {
+			int txPower = setDifferentTxPowerPerAp ? getRandomTxPower() : defaultTxPower;
 			Map<String, Integer> radioMap = new TreeMap<>();
```

Comment:
Maybe just personal preference, I think I prefer
```
txPowerMap
    .computeIfAbsent(serialNumber, k -> new TreeMap<>()).
    put(band, txPower);
```

Reply:
hmm i like the current implementation - it's clearer that the map is created for each loop since the keys are from a map and have to be unique